src/pretix/presale/templates/pretixpresale/fragment_modals.html

```diff
@@ -15,3 +18,95 @@ <h3 id="loadingmodal-label"></h3>
         </div>
     </div>
 </div>
+{% if request.organizer and request.organizer.settings.cookie_consent %}
+    <script type="text/plain" id="cookie-consent-storage-key">cookie-consent-{{ request.organizer.slug }}</script>
+    {% if cookie_consent_from_widget %}
+        {{ cookie_consent_from_widget|json_script:"cookie-consent-from-widget" }}
+    {% endif %}
+    {% if cookie_providers %}
+        <div id="cookie-consent-modal" aria-live="polite">
+            <div class="modal-card">
+                <div class="modal-card-content">
+                    <h3 id="cookie-consent-modal-label"></h3>
+                    <div id="cookie-consent-modal-description">
+                        {% with request.event|default:request.organizer as sh %}
+                            <h3>{{ sh.settings.cookie_consent_dialog_title }}</h3>
+                            {{ sh.settings.cookie_consent_dialog_text|rich_text }}
+                            {% if sh.settings.cookie_consent_dialog_text_secondary %}
+                                <div class="text-muted">
+                                    {{ sh.settings.cookie_consent_dialog_text_secondary|rich_text }}
+                                </div>
+                            {% endif %}
+                            <details id="cookie-consent-details">
+                                <summary>
+                                    <span class="fa fa-fw chevron"></span>
+                                    {% trans "Adjust settings in detail" %}
+                                </summary>
+                                <div class="checkbox">
+                                    <label>
+                                        <input type="checkbox" disabled checked="">
+                                        {% trans "Required cookies" %}<br>
+                                        <span class="text-muted">
+                                            {% trans "Functional cookies (e.g. shopping cart, login, payment, language preference) and technical cookies (e.g. security purposes)" %}
+                                        </span>
+                                    </label>
+                                </div>
+                                {% for cp in cookie_providers %}
+                                    <div class="checkbox">
+                                        <label>
+                                            <input type="checkbox" name="{{ cp.identifier }}">
+                                            {{ cp.provider_name }}<br>
```

Comment:
Hm, this feels a bit strange. I can allow cookies per provider, but then only allow the cookie for all the usage-classes the provider needs/wants/offers? How do I activate a provider that offers „Functionality needed“ and optionally offers „Analytics“? As a plugin-author I would need to register multiple providers. Also, from a user-perspective I would expect this the other way around: group by usage_class and list the providers per usage_class?

So instead I would loop over all requested usage_classes and do (sorry, psudeo-code-filter `|csv_or_json`):

`<input type="checkbox" name="{{ usage_class }}" value="{{ list_of_providers|csv_or_json }}">`

Reply:
One question regarding privacy: if we have e.g. analytics-consent, do we really need to ask for consent again, if we add a different analytics provider? I think this is fair to the user so the user can decide to allow analytics based on where the analytcs-data goes, but to be honest, probably only 0.1% of users will do that.

Reply:
> One question regarding privacy: if we have e.g. analytics-consent, do we really need to ask for consent again, if we add a different analytics provider?

As far as I understand the legal situation, yes.